pkg/cmd/roachtest/tests/mixed_version_backup.go

```diff
@@ -2299,6 +2306,53 @@ func (bc *backupCollection) verifyBackupCollection(
 	return nil
 }
 
+func (bc *backupCollection) verifyOnlineRestore(
+	ctx context.Context,
+	l *logger.Logger,
+	rng *rand.Rand,
+	d *BackupRestoreTestDriver,
+	restoredTables []string,
+	internalSystemJobs bool,
+) ([]tableContents, error) {
+	var downloadJobID int
+	if err := d.testUtils.QueryRow(ctx, rng, `SELECT job_id FROM [SHOW JOBS] WHERE description LIKE '%Background Data Download%' ORDER BY created DESC LIMIT 1`).Scan(&downloadJobID); err != nil {
+		return nil, err
+	}
+
+	if rng.Intn(3) == 0 {
+		// Sometimes wait for the download job to complete before fingerprinting.
+		if err := d.testUtils.waitForJobSuccess(ctx, l, rng, downloadJobID, internalSystemJobs); err != nil {
+			return nil, err
+		}
+	}
+	restoredContents, err := d.computeTableContents(
+		ctx, l, rng, restoredTables, bc.contents, "", /* timestamp */
+	)
+	if err != nil {
+		return nil, fmt.Errorf("backup %s: error loading online restored contents: %w", bc.name, err)
+	}
+
+	// Verify the download job did indeed download all the data.
+	if err := d.testUtils.waitForJobSuccess(ctx, l, rng, downloadJobID, internalSystemJobs); err != nil {
+		return nil, err
+	}
+	randomNode := d.testUtils.RandomNode(rng, d.testUtils.roachNodes)
```

Comment:
why is this gateway random? isn't it a fan out RPC to all stores?

Reply:
the opposite: https://github.com/msbutler/cockroach/blob/ef890345521462bccf29284c9a4fb27b0317e72b/pkg/server/span_stats_server.go#L202

Reply:
i dont like span stats.